ui-manchette/src/components/helpers.ts

```diff
@@ -73,6 +82,29 @@ export const operationalPointsHeight = (
     }
   });
 
+export const computeTimeWindow = (trains: ProjectPathTrainResult[]) => {
+  const { minTime, maxTime } = trains.reduce(
+    (times, train) => {
+      if (train.space_time_curves.length === 0) return times;
+
+      const lastCurve = train.space_time_curves[train.space_time_curves.length - 1];
+      if (lastCurve.times.length === 0) return times;
+
+      const firstPoint = +new Date(train.departure_time);
+      const lastPoint =
+        +new Date(train.departure_time) + lastCurve.times[lastCurve.times.length - 1];
```

Comment:
We can use the `.at(-1)` if you don't think it's bothersome with the typing

Reply:
Yes, done :+1: